assets/js/googlesitekit/notifications/components/common/ActionsCTALinkDismiss.js

```diff
@@ -51,12 +51,12 @@ export default function ActionsCTALinkDismiss( {
 			event.preventDefault();
 		}
 
-		navigateTo( ctaLink );
-
 		await Promise.all( [
 			trackEvents.confirm(),
 			dismissNotification( id, { expiresInSeconds: dismissExpires } ),
 		] );
+
+		navigateTo( ctaLink );
```

Comment:
The reason the above calls are awaited before navigating is to prevent the navigation from interrupting or interfering/preventing those requests. Only situations involving non-full-page navigations should be non-blocking in this way, in which case the `await` can be omitted.

Reply:
Absolutely - as mentioned in [Slack](https://fueled10up.slack.com/archives/CBKKQEBR9/p1724693979015859?thread_ts=1724691993.429919&cid=CBKKQEBR9), this could potentially introduce some problems although when testing, these requests seem to have always been made. 

I've bumped up the priority of #9225 to a P0 and lets try to squeeze that fix in the next release.